editor/src/components/canvas/canvas-strategies/absolute-resize-strategy.tsx

```diff
@@ -0,0 +1,148 @@
+    ) &&
+      interactionState.interactionData.type === 'DRAG' &&
+      interactionState.activeControl.type === 'RESIZE_HANDLE'
+      ? 1
+      : 0
+  },
+  apply: (canvasState, interactionState, sessionState) => {
+    if (
+      interactionState.interactionData.type === 'DRAG' &&
+      interactionState.interactionData.drag != null &&
+      interactionState.activeControl.type === 'RESIZE_HANDLE'
+    ) {
+      const drag = interactionState.interactionData.drag
+      const edgePosition = interactionState.activeControl.edgePosition
+
+      const commandsForSelectedElements = canvasState.selectedElements.flatMap(
+        (selectedElement) => {
+          const element: JSXElement | null = withUnderlyingTarget(
+            selectedElement,
+            canvasState.projectContents,
+            {},
+            canvasState.openFile,
+            null,
+            (_, e) => e,
+          )
+          const elementParentBounds =
+            MetadataUtils.findElementByElementPath(
+              sessionState.startingMetadata, // TODO should this be using the current metadata?
+              selectedElement,
+            )?.specialSizeMeasurements.immediateParentBounds ?? null
+
+          if (element == null) {
+            return []
+          }
+
+          return createResizeCommands(
+            element,
+            selectedElement,
+            edgePosition,
+            drag,
+            elementParentBounds,
+          )
+        },
+      )
+      return [...commandsForSelectedElements, updateHighlightedViews('transient', [])]
+    }
+    // Fallback for when the checks above are not satisfied.
+    return []
+  },
+}
+
+function pinsForEdgePosition(edgePosition: EdgePosition): AbsolutePin[] {
+  let horizontalPins: AbsolutePin[] = []
+  let verticalPins: AbsolutePin[] = []
+
+  if (edgePosition.x === 0) {
+    horizontalPins = ['left', 'width']
+  } else if (edgePosition.x === 1) {
+    horizontalPins = ['right', 'width']
+  }
+
+  if (edgePosition.y === 0) {
+    verticalPins = ['top', 'height']
+  } else if (edgePosition.y === 1) {
+    verticalPins = ['bottom', 'height']
+  }
+
+  return [...horizontalPins, ...verticalPins]
+}
+
+function createResizeCommands(
+  element: JSXElement,
+  selectedElement: ElementPath,
+  edgePosition: EdgePosition,
+  drag: CanvasVector,
+  elementParentBounds: CanvasRectangle | null,
+): AdjustCssLengthProperty[] {
+  const pins = pinsForEdgePosition(edgePosition)
+  return mapDropNulls((pin) => {
+    const horizontal = isHorizontalPoint(
+      // TODO avoid using the loaded FramePoint enum
+      framePointForPinnedProp(pin),
+    )
+    const negative =
+      pin === 'right' ||
+      pin === 'bottom' ||
+      (pin === 'width' && edgePosition.x === 0) ||
+      (pin === 'height' && edgePosition.y === 0)
+    const value = getLayoutProperty(pin, right(element.props), ['style'])
+    if (isRight(value) && value.value != null) {
+      // TODO what to do about missing properties?
```

Comment:
We have a similar problem right now in the absolute move strategy. I have a suspicion that maybe this falls into the same problem category as the escape hatch strategy that we've discussed, and so should be solved as part of that.